src/core/utils/color/ColorValue.ts

```diff
@@ -613,70 +613,50 @@ export class ColorValue {
    */
   private static toHsv(tbgr: number): HsvColor {
     const { r, g, b } = ColorValue.getColors(tbgr);
-    let min = r < g ? r : g;
-    if (b < min) {
-      min = b;
-    }
+    const red = r / 255;
+    const green = g / 255;
+    const blue = b / 255;
+    const cMin = Math.min(red, green, blue);
+    const cMax = Math.max(red, green, blue);
+    const delta = cMax - cMin;
+    let hue = 0;
 
-    let max = r > g ? r : g;
-    if (b > max) {
-      max = b;
+    if (0 === delta) {
+      hue = 0;
+    } else if (red === cMax) {
+      hue = ((green - blue) / delta) % 6;
+    } else if (green === cMax) {
+      hue = (blue - red) / delta + 2;
+    } else {
+      hue = (red - green) / delta + 4;
     }
 
-    /* amount of "blackness" present */
-
-    // const v = Math.floor((max / 255.0) * 100 + 0.5);
-    const v = Number(((max / 255.0) * 100).toFixed(2));
-    const deltaRgb = max - min;
-    // const s = max !== 0.0 ? Math.floor((deltaRgb / max) * 100 + 0.5) : 0;
-    const tempS = max !== 0.0 ? (deltaRgb / max) * 100 : 0;
-    const s = Number(tempS.toFixed(2));
-
-    let h = 0;
-
-    if (s) {
-      const redDistance = (max - r) / deltaRgb;
-      const greenDistance = (max - g) / deltaRgb;
-      const blueDistance = (max - b) / deltaRgb;
-
-      let intermediateHue: number;
-      if (r === max) {
-        /* color between yellow & magenta */
-        intermediateHue = blueDistance - greenDistance;
-      } else if (g === max) {
-        /* color between cyan & yellow */
-        intermediateHue = 2.0 + redDistance - blueDistance;
-      } else {
-        /* color between magenta & cyan */
-        intermediateHue = 4.0 + greenDistance - redDistance;
-      }
-
-      /* intermediate hue is [0..6] */
-      intermediateHue *= 60;
-
-      if (intermediateHue < 0.0) {
-        intermediateHue += 360;
-      }
+    hue = Math.round(hue * 60);
+    if (hue < 0) {
+      hue += 360;
+    }
 
-      // h = Math.floor(intermediateHue + 0.5);
-      h = Number(intermediateHue.toFixed(2));
+    let brightness = cMax;
+    let saturation = cMax === 0 ? 0 : delta / cMax;
 
-      if (h >= 360) {
-        h = 0;
-      }
-    } else {
-      h = 0;
-    }
+    // round values to 1 decimal place
+    saturation = Number((saturation * 100).toFixed(1));
+    brightness = Number((brightness * 100).toFixed(1));
 
-    return { h, s, v, a: this.getAlpha(tbgr) / 255 };
+    return {
+      h: hue,
+      s: saturation,
+      v: brightness,
+      a: this.getAlpha(tbgr) / 255,
+    };
   }
 
   /** True if the value of this ColorValue is the same as another ColorValue. */
   public equals(other: ColorValue): boolean {
     return this._tbgr === other._tbgr && this._hue === other._hue;
   }
 
-  public static getFormattedColorNumber(value: number, precision = 2) {
+  public static getFormattedColorNumber(value: number, precision = 1) {
```

Comment:
I don't think alpha values should be rounding to 1 decimal place because the alpha slider is kind of mess up now

![alphavalue](https://user-images.githubusercontent.com/88331924/141329329-c9316ee6-9219-434f-85ca-940c3c07c683.gif)

Reply:
Yes, you are correct ... I will fix that

Reply:
Fixed